zerver/lib/alert_words.py

```diff
@@ -36,26 +34,31 @@ def get_alert_word_automaton(realm: Realm) -> ahocorasick.Automaton:
     return alert_word_automaton
 
 def user_alert_words(user_profile: UserProfile) -> List[str]:
-    return ujson.loads(user_profile.alert_words)
+    alert_word_list = []
+    for alert in AlertWord.objects.filter(user_profile=user_profile):
+        alert_word_list.append(alert.word)
+    return alert_word_list
 
 def add_user_alert_words(user_profile: UserProfile, alert_words: Iterable[str]) -> List[str]:
     words = user_alert_words(user_profile)
 
     new_words = [w for w in alert_words if w not in words]
-    words.extend(new_words)
 
-    set_user_alert_words(user_profile, words)
+    AlertWord.objects.bulk_create(
+        AlertWord(user_profile=user_profile, word=word, realm=user_profile.realm) for word in new_words)
 
-    return words
+    return words+new_words
 
 def remove_user_alert_words(user_profile: UserProfile, alert_words: Iterable[str]) -> List[str]:
     words = user_alert_words(user_profile)
-    words = [w for w in words if w not in alert_words]
+    delete_words = [w for w in alert_words if w in words]
+    for word in delete_words:
+        w = AlertWord.objects.get(user_profile=user_profile, word=word)
+        w.delete()
```

Comment:
I think this can be a bulk Django query, e.g. `AlertWord.objects.filter(user_profile=user_profile, word__in=[alert_words]).delete()`.